.github/workflows/upload.yml

```diff
@@ -2,21 +2,22 @@ name: upload
 
 on:
   push:
-    branches:
-      - master
+    tags:
+      - 'v*.*.*'
```

Comment:
why not using master?

Reply:
I guess, by doing such, we could control when to release by adding a tag, instead of whenever there is a MR being merged to master branch.